junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -129,19 +132,18 @@ boolean hasException() {
 
     void notifyException(Exception exception) {
         this.exception = exception;
-        // Commented because of issue https://github.com/k3po/k3po/issues/391
-//        prepared.countDown();
-//        startable.countDown();
-//        finished.countDown();
-        if (testThread != null) {
-            if (testThread != Thread.currentThread())
-                testThread.interrupt();
-        }
+        interruptTestThread();
```

Comment:
I'd like to see the code further simplified if possible please.

If calling `Thread.interrupt()` on an already interrupted thread is a no-op, then I'd suggest the following instead...
```
if (exception != null && !currentThread().equals(testThread)) {
    testThread.interrupt();
}
```
...and remove the `testThreadInterrupted` field and the separate `interruptTestThread()` method.

Note: this is common practice to move the possibly-null variable to the argument of `equals` so that a `null` value returns `false` for `equals` without requiring the extra `!= null` check.

For example:
`"nonNullString".equals(nullableString)`
instead of
`nullableString != null && nullableString.equals("nonNullString")`

Note: also common practice to put the cheap-check first in an `if` condition:
`if (exception != null && !currentThread().equals(testThread)) ...`
instead of
`if (!currentThread().equals(testThread) && exception != null) ...`